### PR TITLE
Notify CredentialsUpdated event when password is changed

### LINKED ISSUE #33: ICredentialsUpdatedEvent is never generated
The password seems to be changed here https://github.com/plone/Products.PlonePAS/blob/master/src/Products/PlonePAS/plugins/user.py#L82 but I can't see anywhere the event is sent and no subscriber seems to receive the event.

I can't see any tests for this event either.

### PR DESCRIPTION
**Pull request by ericof**
Fixes #33

### CODE REVIEW

**Comment by ericof** [comment]
@jenkins-plone-org please run jobs

File: src/Products/PlonePAS/tests/test_membershiptool.py [hunk 1]
```diff
@@ -700,6 +704,26 @@ def testGetBadMembers(self):
...
+
+        self.assertTrue(self.membership.testCurrentPassword('secret'))
+        self.assertFalse(self.membership.testCurrentPassword('whoknows'))
+        login(self.portal, TEST_USER_NAME)  # Back to normal
+        self.membership.setPassword('guessagain')
+        self.assertEqual(len(events_fired), 1)
+        self.assertEqual(events_fired[0].principal.getId(), TEST_USER_ID)
+        self.assertEqual(events_fired[0].password, 'guessagain')
+
+        gsm.unregisterHandler(got_credentials_updated_event)
```
Comment (jensens) [inline src/Products/PlonePAS/tests/test_membershiptool.py]: Do we need to unregister? Anyway, it does not hurt, but the test-layer should take care of it.

**Comment by ericof** [comment]
@jenkins-plone-org please run jobs

File: src/Products/PlonePAS/tests/test_membershiptool.py (hunk 1 above)
Comment (ericof) [reply src/Products/PlonePAS/tests/test_membershiptool.py]: Done!

**Pull request merged**